streams/src/main/java/org/apache/kafka/streams/state/internals/TimeOrderedKeyValueBuffer.java

```diff
@@ -27,14 +26,14 @@
 import java.util.function.Consumer;
 import java.util.function.Supplier;
 
-public interface TimeOrderedKeyValueBuffer<K, V> extends StateStore {
+public interface TimeOrderedKeyValueBuffer<K, V, T> extends StateStore {
```

Comment:
Can we add a quick javadoc to clarify that `T` here is the buffer type? Should already be clear to anyone who reads the code, but a javadoc will help make it so readers won't need to read the code if all they want to know is what the type is.